src/alignment/Graph.cpp

```diff
@@ -1,124 +1,141 @@
-			if (i+kmer > stringList.at(seq+1).size())
-				break;                         
-			if (stringList.at(seq).substr(position,kmer) == stringList.at(seq+1).substr(i,kmer))
-				adjacentNode.push_back(i/kmer);		
-		}
-		return adjacentNode;
-	} else {
-		cerr << kNum << ".k-mer don't exists in sequence " << seq  << "." << endl;
-		return adjacentNode; 
-	}	
-	cerr << kNum << ".k-mer don't exists in sequence " << seq  << "." << endl;
-	return adjacentNode;
+    // the last sequence has no next sequence
+    if (index == stringList.size()-1) {                                 
+        cerr << "It is the last sequence." << endl;
+        return adjacentNodes;
+    }
+    // calculate adjacent edges from all nodes in one sequence 
+    for (int i = 0; i < nodeList.at(index).size(); i++) {
+        // initialize adjacentNode
+        adjacentNodes.push_back(emptyVecNode);
+        for (int j = 0; j < nodeList.at(index + 1).size(); j++) {
+            // compare strings of nodes
+            if (nodeList.at(index).at(i).kmer == nodeList.at(index + 1).at(j).kmer){
+                adjacentNodes.at(i).push_back(nodeList.at(index + 1).at(j));  
+            }            
+        }
+    }
+    return adjacentNodes;
 }
-// Method: get Number of k-mer
+
+// Method: get amount of nodes for all sequences
 const vector<int>& Graph::getAmountOfKmer(){
-	vector<string>& stringList = getStringListSequence();
-	int sizeOfStringList = stringList.size();
+    vector<string>& stringList = getStringListSequence();
+    int sizeOfStringList = stringList.size();
 
-	if (!numberOfKmer.empty()) {
-		return numberOfKmer; 		
-	} else {
-		for (int i = 0; i < sizeOfStringList; i++) {
-			numberOfKmer.push_back(stringList.at(i).size() / getK());						
-		}
-	}
-	return numberOfKmer; 
+    if (!numberOfKmers.empty()) {
+        return numberOfKmers; 		
+    } else {
+        for (int i = 0; i < sizeOfStringList; i++) {
+            numberOfKmers.push_back(stringList.at(i).size() / getK());						
+        }
+    }
+    return numberOfKmers; 
 }					
 
 // Method: list of edges
 void Graph::calcEdgeList() {
-	vector<string>& stringList = getStringListSequence();
-	const vector<int>& amountKmer = getAmountOfKmer();
-	vector<int> adjEdges;
-	int j;
+    // calculate list of nodes
+    calcNodeList();
 
-	for (int i = 0; i < stringList.size()-1; i++) {
-		for (j = 0; j < amountKmer.at(i); j++) {
-			adjEdges = getAdjacentEdge(i,j);
+    // fill edge list
+    for (int i = 0; i < nodeList.size() - 1; i++) {      
+            listOfEdges.push_back(getAdjacentEdge(i));
+            }
+        }	
 
-			for (auto &k : adjEdges) {
-			array<int,3> edge = {i,j,k};
-			listOfEdges.push_back(edge);
-			}
-		}	
-	}
+// Method:: calculate list of nodes
+void Graph::calcNodeList() {
+    vector<string>& stringList = getStringListSequence();
+    const vector<int>& amountKmer = getAmountOfKmer();
+    int stringLength = k;
+    int j;

```

Comment:
```suggestion
    uint j;
```
The compiler gives a warning `comparison between signed and unsigned integer expressions `.